### PR TITLE
Giant refractor.

### PR DESCRIPTION
**Pull request by iTitus**

### CODE REVIEW

**Comment by iTitus** [comment]
This had to be done

**Comment by iTitus** [comment]
/me approves all the approvals and hopes @AEnterprise does so too.

File: src/main/java/buildcraftAdditions/ModIntegration/Framez/MovementHandler.java [hunk 1]
```diff
@@ -1,14 +1,15 @@
 package buildcraftAdditions.ModIntegration.Framez;
 
+import com.amadornes.framez.api.movement.BlockMovementType;
+import com.amadornes.framez.api.movement.IMovementHandler;
+import com.amadornes.framez.api.movement.IMovingBlock;
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/ModIntegration/Framez/MovementHandler.java]: i think your import settings are wrong here or is was this my intelliJ derping?

File: src/main/java/buildcraftAdditions/ModIntegration/ModIntegration.java [hunk 2]
```diff
@@ -71,25 +71,25 @@ private static void railcraftIntegration() {
 	private static void eurekaResearch() {
 		EurekaRegistry.registerCategory("BCA", new ItemStack(ItemsAndBlocks.kineticDusterBlock));
 
-		EurekaRegistry.register(new EurekaInfo(Variables.DustT0Key, "BCA", 1, new ItemStack(ItemsAndBlocks.basicDusterBlock)));
-		EurekaRegistry.registerDrops(Variables.DustT0Key, new ItemStack(eureka.utils.BCItems.STONE_GEAR, 2), new ItemStack(Items.iron_ingot), new ItemStack(Blocks.stone, 5), new ItemStack(Items.slime_ball));
-		EurekaRegistry.bindToKey(ItemsAndBlocks.basicDusterBlock, Variables.DustT0Key);
+		EurekaRegistry.register(new EurekaInfo(Variables.Eureka.DustT0Key, "BCA", 1, new ItemStack(ItemsAndBlocks.basicDusterBlock)));
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/ModIntegration/ModIntegration.java]: told ya to don't touch this stuff, rewriting eureka so redoing this anyways

File: src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java [hunk 3]
```diff
@@ -19,10 +20,12 @@
  * Please check the contents of the license located in
  * http://buildcraftadditions.wordpress.com/wiki/licensing-stuff/
  */
-public class KineticBackpack extends ItemArmor implements IEnergyContainerItem {
+public class ItemKineticBackpack extends ItemArmor implements IEnergyContainerItem {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java]: this is also something you shouldn't have touched, i had local changes to this, i'm working on this in between other things

File: src/main/java/buildcraftAdditions/blocks/BlockBackpackStand.java [hunk 4]
```diff
@@ -16,22 +17,21 @@
...
 	public BlockBackpackStand() {
-		super();
-		setBlockName("BlockBackpackStand");
+		super("backpackStand");
 	}
 
 	@Override
-	public void onBlockPlacedBy(World p_149689_1_, int p_149689_2_, int p_149689_3_, int p_149689_4_, EntityLivingBase p_149689_5_, ItemStack p_149689_6_) {
-		super.onBlockPlacedBy(p_149689_1_, p_149689_2_, p_149689_3_, p_149689_4_, p_149689_5_, p_149689_6_);
+	public void onBlockPlacedBy(World world, int x, int y, int z, EntityLivingBase entity, ItemStack stack) {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/blocks/BlockBackpackStand.java]: same as the kinetic backpack, work in progress, was already changed locally but not pushed, this was just to test what functions returned what

File: src/main/java/buildcraftAdditions/blocks/BlockTest.java [hunk 5]
```diff
@@ -17,10 +18,13 @@
...
  * http://buildcraftadditions.wordpress.com/wiki/licensing-stuff/
  */
-public class PaternTestingBlock extends Block {
+public class BlockTest extends Block {
 
-	public PaternTestingBlock() {
+	public BlockTest() {
 		super(Material.iron);
+		setBlockName("testBlock");
+		setBlockTextureName("bcadditions:testBlock");
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/blocks/BlockTest.java]: this doesn't need a texture

File: src/main/java/buildcraftAdditions/client/gui/GuiBase.java [hunk 6]
```diff
@@ -31,11 +32,10 @@
 public abstract class GuiBase extends GuiContainer {
 
 	public static final ResourceLocation PLAYER_INV_TEXTURE = new ResourceLocation("bcadditions:textures/gui/guiPlayerInv.png");
-
+	public final List<WidgetBase> widgets = new ArrayList<WidgetBase>();
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/client/gui/GuiBase.java]: this list CAN NOT be final!

File: src/main/java/buildcraftAdditions/items/Tools/ItemKineticMultiTool.java [hunk 7]
```diff
@@ -80,6 +80,150 @@ public ItemKineticMultiTool() {
 		setNoRepair();
 	}
 
+	public static void setLastUsedMode(ItemStack stack, String mode) {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/items/Tools/ItemKineticMultiTool.java]: what's the point in moving arround this code?

File: src/main/java/buildcraftAdditions/multiBlocks/IMultiBlockTile.java [hunk 8]
```diff
@@ -28,25 +29,25 @@
 
 	public int getMasterX();
 
+	public void setMasterX(int masterX);
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/multiBlocks/IMultiBlockTile.java]: what's the point in changing order here?

File: src/main/java/buildcraftAdditions/proxy/CommonProxy.java [hunk 9]
```diff
@@ -24,7 +52,7 @@
  * http://buildcraftadditions.wordpress.com/wiki/licensing-stuff/
  */
 
-public class CommonProxy {
+public class CommonProxy implements IGuiHandler {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/proxy/CommonProxy.java]: why are you putting this here????

File: src/main/java/buildcraftAdditions/tileEntities/TileCoolingTower.java [hunk 10]
```diff
@@ -230,43 +230,48 @@ public int getMasterX() {
...
+
+	@Override
 	public int getMasterY() {
 		return data.masterY;
 	}
 
 	@Override
-	public int getMasterZ() {
-		return data.masterZ;
+	public void setMasterY(int masterY) {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/tileEntities/TileCoolingTower.java]: same here, what's the point in changing the order of the functions?

File: src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java [hunk 11]
```diff
@@ -35,11 +35,11 @@
  * http://buildcraftadditions.wordpress.com/wiki/licensing-stuff/
  */
 public class TileHeatedFurnace extends TileBase implements ISidedInventory, IUpgradableMachine, IConfigurableOutput {
+	public final TileCoilBase[] coils;
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java]: can't be final!

Comment (iTitus) [inline src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java]: You can actually change the contetnts of an array when its final :P

File: src/main/java/buildcraftAdditions/client/gui/GuiBase.java (hunk 6 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/client/gui/GuiBase.java]: You can actually change the contetnts of a list when its final :P

File: src/main/java/buildcraftAdditions/items/Tools/ItemKineticMultiTool.java (hunk 7 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/items/Tools/ItemKineticMultiTool.java]: IntelliJ...

File: src/main/java/buildcraftAdditions/proxy/CommonProxy.java (hunk 9 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/proxy/CommonProxy.java]: Clean code?
The only point of this proxy class is the differentiation between server and client. A GUIHandler ineeds client/server separation. Its the perfect use of the proxy classes...

File: src/main/java/buildcraftAdditions/blocks/BlockTest.java (hunk 5 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/blocks/BlockTest.java]: Just wanted to get rid of those Exceptions in the console to filter out Execptions which should be fixed

File: src/main/java/buildcraftAdditions/ModIntegration/Framez/MovementHandler.java (hunk 1 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/ModIntegration/Framez/MovementHandler.java]: import com. is on the top.
Maybe i should make com.amadornes.* a low priority.

File: src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java (hunk 3 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java]: Shall I revert my changes here?

Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java]: mine are pretty minor and just testing so i'll revert mine, doesn't matter much, just reminding you that you should watch out for things like that

File: src/main/java/buildcraftAdditions/client/gui/GuiBase.java (hunk 6 above)
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/client/gui/GuiBase.java]: last time i tried something like that i got errors though

File: src/main/java/buildcraftAdditions/proxy/CommonProxy.java (hunk 9 above)
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/proxy/CommonProxy.java]: makes more sense to me to have a guiHandler

File: src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java (hunk 11 above)
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java]: that doesn't make sense though

File: src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java (hunk 3 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java]: Sorry, I always forget of the Jenkins. I think that everybody has pushed their changes...

File: src/main/java/buildcraftAdditions/proxy/CommonProxy.java (hunk 9 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/proxy/CommonProxy.java]: Will revert.

File: src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java (hunk 11 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/tileEntities/TileHeatedFurnace.java]: #BlameJava

**Comment by iTitus** [comment]
I think I found some issues in the duster code that could cause progress loss...

File: src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java (hunk 3 above)
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/armour/ItemKineticBackpack.java]: i mostly push them but not while i'm testing out new stuff, especially if i don't know yet if it works or not

**Comment by AEnterprise** [comment]
merged yesterday, git derped

**Pull request closed without merging**